src/TraceEvent/Computers/TraceManagedProcess.cs

```diff
@@ -244,6 +251,36 @@ internal static void SetupCallbacks(TraceEventDispatcher source)
             clrRundownParser.RuntimeStart += doAtRuntimeStart;
             source.Clr.RuntimeStart += doAtRuntimeStart;
 
+            var symbolParser = new SymbolTraceEventParser(source);
+            symbolParser.ImageIDFileVersion += delegate (FileVersionTraceData data)
+            {
+                TraceProcess process = data.Process();
+
+                if (!string.IsNullOrWhiteSpace(data.OrigFileName) &&
+                    (data.OrigFileName.Equals("clr.dll", StringComparison.OrdinalIgnoreCase) || data.OrigFileName.Equals("mscorwks.dll", StringComparison.OrdinalIgnoreCase))
+                    )
+                {
+                    TraceLoadedDotNetRuntime mang = currentManagedProcess(data); // this will create a mang instance for this process
+                    var version = Version.Parse(data.ProductVersion);
+                    // replace the current runtimeVersion if there is not good revision information
+                    if (mang.runtimeVersion.Major == 0 || mang.runtimeVersion.Revision == 0) mang.runtimeVersion = new Version(version.Major, version.Minor, version.Build, version.Revision);
+                    if (mang.RuntimeBuiltTime == default(DateTime)) mang.RuntimeBuiltTime = data.BuildTime;
+                }
+            };
+            symbolParser.ImageID += delegate (ImageIDTraceData data)
+            {
+                TraceProcess process = data.Process();
+
+                if (!string.IsNullOrWhiteSpace(data.OriginalFileName) &&
```

Comment:
:bulb: Use `string.Equals(data.OriginalFileName, "clr.dll", StringComparison.OrdinalIgnoreCase)`. It eliminates the need for the null/empty/whitespace check, and also for this case eliminates the need to have both `&&` and `||` in the same expression.

:memo: This occurs in another case above as well.